File: java/test/src/main/java/test/Ice/invoke/ServantLocatorI.java
// Copyright (c) ZeroC, Inc.

package test.Ice.invoke;

import com.zeroc.Ice.Current;
import com.zeroc.Ice.Object;
import com.zeroc.Ice.ServantLocator;

public class ServantLocatorI implements ServantLocator {
    public ServantLocatorI(boolean async) {
        _blobject = new ServantLocator.LocateResult();
        if (async) {
            _blobject.returnValue = new BlobjectAsyncI();
        } else {
            _blobject.returnValue = new BlobjectI();
        }
    }

    @Override
    public ServantLocator.LocateResult locate(Current current) {
        return _blobject;
    }

    @Override
    public void finished(
            Current current, Object servant, java.lang.Object cookie) {}

    @Override
    public void deactivate(String category) {}

    private final ServantLocator.LocateResult _blobject;
}

File: swift/test/Ice/exceptions/ServantLocatorI.swift
import Ice

class ServantLocatorI: Ice.ServantLocator {

    func locate(_ curr: Current) throws -> (returnValue: Object?, cookie: AnyObject?) {
        return (nil, nil)
    }

    func finished(curr: Current, servant: Object, cookie: AnyObject) throws {
    }

    func deactivate(_ category: String) {
    }
}
